src/table-viewer.ts

```diff
@@ -48,7 +56,12 @@ export class TableViewer extends cdk.Construct {
 
     props.table.grantReadData(handler);
 
-    const home = new apigw.LambdaRestApi(this, 'ViewerEndpoint', { handler });
+    const home = new apigw.LambdaRestApi(this, 'ViewerEndpoint', {
+      handler,
+      endpointConfiguration: {
+        types: [props.endpointType || apigw.EndpointType.EDGE],
+      },
```

Comment:
Let's leave the current behavior if not set:

```suggestion
      endpointConfiguration: props.endpointType 
  ? { types: [props.endpointType] }
  : undefined,
```

Reply:
Updated